File: ios/RNFollowAnalyticsGDPR.m
#import "RNFollowAnalyticsGDPR.h"
#import <React/RCTLog.h>

@implementation RNFollowAnalyticsGDPR

- (dispatch_queue_t)methodQueue
{
    return dispatch_get_main_queue();
}
RCT_EXPORT_MODULE()
// GDPR
RCT_EXPORT_METHOD(requestToAccessMyData)
{
    [FollowAnalytics.GDPR requestToAccessMyData];
}

RCT_EXPORT_METHOD(requestToDeleteMyData)
{
    [FollowAnalytics.GDPR requestToDeleteMyData];
}

@end
  

File: android/src/main/java/com/followanalytics/reactnative/sdk/RNFollowAnalyticsGDPR.java
package com.followanalytics.reactnative.sdk;

import android.util.Log;

import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;
import com.facebook.react.bridge.Callback;
import com.followanalytics.FollowAnalytics;

public class RNFollowAnalyticsGDPR extends ReactContextBaseJavaModule  {
    private final ReactApplicationContext reactContext;

    public RNFollowAnalyticsGDPR(ReactApplicationContext reactContext) {
        super(reactContext);
        this.reactContext = reactContext;
    }

    @Override
    public String getName() {
        return "RNFollowAnalyticsGDPR";
    }

    @ReactMethod
    public void requestAccessToMyData() {
        try {
            FollowAnalytics.GDPR.requestToAccessMyData();
        } catch (Exception e) {
            Log.d("Error requesting access", "");
        }
    }

    @ReactMethod
    public void requestToDeleteMyData() {
        try {
            FollowAnalytics.GDPR.requestToDeleteMyData();
        } catch (Exception e) {
            Log.d("Error requesting access", "");
        }
    }
}
